test/unit/collection.test.js

```diff
@@ -1,29 +1,19 @@
 'use strict';
 
-const EventEmitter = require('events');
 const chai = require('chai');
 const expect = chai.expect;
-const { Db } = require('../../src/db');
-
-class MockTopology extends EventEmitter {
-  constructor() {
-    super();
-  }
-
-  capabilities() {
-    return {};
-  }
-}
+const { withClient } = require('../functional/shared');
 
 describe('Collection', function () {
   it('should not allow atomic operators for findOneAndReplace', {
     metadata: { requires: { topology: 'single' } },
-    test: function () {
-      const db = new Db('fakeDb', new MockTopology());
+    test: withClient((client, done) => {
+      const db = client.db('fakeDb');
       const collection = db.collection('test');
       expect(() => {
         collection.findOneAndReplace({ a: 1 }, { $set: { a: 14 } });
       }).to.throw(/must not contain atomic operators/);
-    }
+      done();
+    })
   });
```

Comment:
This test needed to change because now the `Db` constructor takes a `MongoClient` instead of a `topology`. Perhaps this test could/should be moved to `test/functional`?

Reply:
Yeah makes sense.